n0tl3ss/micronaut-oracle-cloud
# Remove use of non-public and deprecated GraalVM API

- a
- cleanup generation code
- Disable transitive dependencies
- Fix #386 (#388)
- [skip ci] Release v2.1.5
- Back to 2.1.6-SNAPSHOT
- Upgrade OCI SDK to 2.36.1
- Upgrade OCI SDK to 2.37.0
- Upgrade OCI SDK to 2.37.0
- add Oracle Cloud Log feature (#394)
- minor logging improvements
- add --no-build-cache to gradle workflow
- fix checkstyle for http metadata url
- catch general exception
- tryToConfigure when is succesfull dont have to configure it again
- fix appender
- fix micrometer test
- Upgrade OCI SDK to 2.41.0
- fix checkstyle
- add --no-build-cache to gradle.yml
- add missing com.oracle.bmc.identity.SdkClientsMetadata
- "test it publish metrics to ingestion telemetry endpoint" requires env variable to be ran
- Fix tests
- Fix tests
- Fix tests
- send logs into batches
- update GraalVM to 22.2.0
- update GraalVM to 22.1.0
- Cross reference to Micronaut Object Storage
- Update gradle.properties
- [skip ci] Release v2.2.0
- Back to 2.2.1-SNAPSHOT
- Update cloudStorage.adoc
- switch from jackson to micronaut serialization
- [skip ci] Release v2.2.1
- Back to 2.2.2-SNAPSHOT
- doc: Add breaks.adoc mentioning transitive dependency removal (#408)
- Upgrade build
- Address review feedback
- Remove discovery dependency from OCI logging
- Fix tests
- OCI SDK 3 compatibility, add netty-based OCI http client This PR adds compatibility with OCI SDK 3.0.0. The changes for compatibility are minor. We can't upgrade to OCI SDK 3 until Micronaut 4, but this change will let users upgrade their SDK if manually.
- Fix tests
- spotless
- Fix tests
- sonar found an actual bug!
- disable japicmp for now
- specific exception for sonar
- fix assertion order
- fix test
- test
- implement BUFFER_REQUEST
- Disable catalog check on Java 8
- fix native-image build
- implement tls settings
- spotless
- revert sdk version
- fix java 8 compat
- move to new 22.3 graal public api for RuntimeJNIAccess
- revert gh actions gradle version
- spotless
- use reflection to access new graal apis
- Fix #404 wrong property check
- sonar
- update core branch
- update project version
- [skip ci] Release v2.3.0
- Back to 2.3.1-SNAPSHOT
- GraalVM OCI runners
- GraalVM OCI runners
- GraalVM OCI runners
- Pipelines scirpt
- GraalVM OCI runners
- GraalVM OCI runners
- GraalVM OCI runners
- GraalVM OCI runners
- GraalVM OCI runners
- GraalVM OCI runners
- GraalVM OCI runners
- GraalVM OCI runners
- GraalVM OCI runners
- GraalVM OCI runners
- GraalVM OCI runners
- Fix oci sdk 3 binary compatibility ClientConfigurationBuilder has a reference to a class (JaxRsCircuitBreaker) that is gone in oci sdk 3. This lead to a NoClassDefFoundError in the generated code that loads the config. You can't set this field from config anyway: Its type is an interface.
- exclude transitive to avoid version conflict
- mock providers to fix running without ~/.oci/config present
- use java.util.Supplier
- fix another version divergence...
- [skip ci] Release v2.3.1
- Back to 2.3.2-SNAPSHOT
- Bump mikepenz/action-junit-report from 3.0.3 to 3.6.2 (#448)
- Bump stefanzweifel/git-auto-commit-action from 4.13.1 to 4.16.0 (#447)
- Bump peter-evans/create-pull-request from 3.12.1 to 4.2.3 (#443)
- Bump actions/cache from 2.1.7 to 3.0.11 (#422)
- Bump actions/setup-java from 2 to 3 (#368)
- Bump mikepenz/action-junit-report from 3.6.2 to 3.7.0 (#451)
- Bump AutoModality/action-clean from 1.1.0 to 1.1.1 (#450)
- [oracle-cloud] Update common files for branch master (#344)
- fix: use @Nullable instead of Optional in constructor parameter (#410)
- Upgrade to Micronaut 4.0.x (#453)
- Update ociVersion to v3.2.1 (#465)
- Update mikepenz/action-junit-report action to v3.7.1 (#470)
- Update dependency org.eclipse.jetty:jetty-server to v9.4.50.v20221201 (#456)
- Update dependency com.oracle.database.jdbc:ojdbc-bom to v21.8.0.0 (#459)
- Update dependency org.eclipse.jetty:jetty-server to v11 (#466)
- Update kotlin monorepo to v1.8.0 (#462)
- Update dependency org.slf4j:jcl-over-slf4j to v2 (#467)
- Update dependency org.mockito:mockito-core to v5 (#472)
- Update plugin io.micronaut.build.shared.settings to v6.2.0 (#469)
- fix(deps): update dependency io.micronaut.gradle:micronaut-gradle-plugin to v3.7.0 (#455)
- [oracle-cloud] Update common files for branch master (#454)
- Clean versions (#473)
- [oracle-cloud] Update common files for branch master (#474)
- fix(deps): update dependency org.graalvm.nativeimage:svm to v22.3.1 (#477)
- Merge 2.3.x into master (#480)
- [oracle-cloud] Update common files for branch master (#481)
- fix(deps): update dependency org.mockito:mockito-core to v5.1.0 (#485)
- fix(deps): update oci to v3.2.2 (#484)
- chore(deps): update plugin io.micronaut.build.shared.settings to v6.2.1 (#483)
- [oracle-cloud] Update common files for branch master (#486)
- fix(deps): update oci to v3.3.0 (#494)
- fix(deps): update ojdbc to v21.9.0.0 (#499)
- fix(deps): update fn to v1.0.165 (#498)
- chore(deps): update plugin io.micronaut.build.shared.settings to v6.2.2 (#497)
- [oracle-cloud] Update common files for branch master (#487)
- chore(deps): update gradle/gradle-build-action action to v2.4.0 (#506)
- fix(deps): update oci to v3.4.0 (#504)
- fix(deps): update dependency io.micronaut.gradle:micronaut-gradle-plugin to v3.7.2 (#503)
- build: plugin 6.3.5, gradle 8.0.2, jakarta validation, standardized project names, cleanup (#518)
- Use Milestones (#520)
- [skip ci] Release v3.0.0-M1
- Back to 3.0.0-SNAPSHOT
- Remove use of non-public and deprecated GraalVM API
